src/renderer/components/cluster-manager/cluster-view.tsx

```diff
@@ -68,6 +68,11 @@ class NonInjectedClusterView extends React.Component<Dependencies> {
   bindEvents() {
     disposeOnUnmount(this, [
       reaction(() => this.clusterId, async (clusterId) => {
+        // TODO: replace with better handling
+        if (clusterId && !this.props.entityRegistry.getById(clusterId)) {
```

Comment:
```suggestion
        if (!clusterId || !this.props.entityRegistry.getById(clusterId)) {
```
You don't want it activating if `clusterId` is undefined, right? Although I guess that is the pre-existing behaviour...

Reply:
I think we do because that clears the current cluster value on main